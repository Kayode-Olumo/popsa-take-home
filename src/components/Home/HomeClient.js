"use client";

import Image from "next/image";
import Link from "next/link";
import Intro from "@/app/styles/home.styles";
import CommonStrings from "@/src/utils/constants";

export default function HomeClient() {
  return (
    <Intro>
      <Image src="/logo.svg" alt="Popsa" width="127" height="32" />
      <h1>{CommonStrings.mainPage.intro}</h1>
      <Link href="/gallerypage">{CommonStrings.mainPage.link}</Link>
    </Intro>
  );
}
