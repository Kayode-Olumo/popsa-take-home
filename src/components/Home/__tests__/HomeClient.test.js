import { render, screen } from "@testing-library/react";
import HomeClient from "@/src/components/Home/HomeClient";
import CommonStrings from "@/src/utils/constants";

// Mock next/image since it's not available in test environment
jest.mock("next/image", () => ({
  __esModule: true,
  default: (props) => <img {...props} />,
}));

// Mock next/link
jest.mock("next/link", () => ({
  __esModule: true,
  default: ({ children, href }) => <a href={href}>{children}</a>,
}));

describe("HomeClient", () => {
  it("renders the logo", () => {
    render(<HomeClient />);
    const logo = screen.getByAltText("Popsa");
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute("width", "127");
    expect(logo).toHaveAttribute("height", "32");
  });

  it("displays the welcome message", () => {
    render(<HomeClient />);
    const heading = screen.getByRole("heading", {
      name: CommonStrings.mainPage.intro,
    });
    expect(heading).toBeInTheDocument();
  });

  it("contains a link to start the test", () => {
    render(<HomeClient />);
    const link = screen.getByRole("link", {
      name: CommonStrings.mainPage.link,
    });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/gallerypage");
  });

  it("matches snapshot", () => {
    const { container } = render(<HomeClient />);
    expect(container).toMatchSnapshot();
  });
});
