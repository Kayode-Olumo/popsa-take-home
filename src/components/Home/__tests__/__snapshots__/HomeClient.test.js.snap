// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeClient matches snapshot 1`] = `
<div>
  <div
    class="sc-34dd1efc-0 cbRrnm"
  >
    <img
      alt="Popsa"
      height="32"
      src="/logo.svg"
      width="127"
    />
    <h1>
      Welcome to the React frontend test
    </h1>
    <a
      href="/gallerypage"
    >
      Start test
    </a>
  </div>
</div>
`;
